lib/streamlit/testing/element_tree.py

```diff
@@ -549,31 +549,42 @@ def __init__(self, proto: NumberInputProto, root: ElementTree):
         self.disabled = proto.disabled
         self.key = user_key_from_widget_id(self.id)
 
-    def set_value(self, v: Number) -> NumberInput:
+    def set_value(self, v: Number | None) -> NumberInput:
         self._value = v
         return self
 
     def widget_state(self) -> WidgetState:
         ws = WidgetState()
         ws.id = self.id
-        ws.double_value = self.value
+        if self.value is not None:
+            ws.double_value = self.value
         return ws
 
     @property
-    def value(self) -> Number:
+    def value(self) -> Number | None:
         if self._value is not None:
             return self._value
         else:
             state = self.root.session_state
             assert state
+
+            if self.id not in state:
```

Comment:
@AnOctopus  Would this part work fine?

Reply:
If this is not correct, we might need to also introduce some kind of default class as a state instead of using `None` for `_value`